x-pack/plugins/endpoint/public/embeddables/resolver/store/data/selectors.ts

```diff
@@ -0,0 +1,302 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { createSelector } from 'reselect';
+import { ResolverState } from '../../types';
+import { sampleData } from './sample';
+import { depthFirstPreorder, levelOrder } from '../../lib/tree_sequencers';
+import { Vector2 } from '../../types';
+import { add as vector2Add } from '../../lib/vector2';
+
+const unit = 100;
+const distanceBetweenNodesInUnits = 1;
+const distanceBetweenNodes = distanceBetweenNodesInUnits * unit;
+
+function dataSelector(state: ResolverState) {
+  return sampleData.data.result.search_results;
+}
+
+export function isGraphableProcess(event) {
+  return eventType(event) === 'processCreated' || eventType(event) === 'processRan';
+}
+
+export function eventType(event) {
+  const {
+    data_buffer: { event_type_full: type, event_subtype_full: subType },
+  } = event;
+
+  if (type === 'process_event') {
+    if (subType === 'creation_event' || subType === 'fork_event' || subType === 'exec_event') {
+      return 'processCreated';
+    } else if (subType === 'already_running') {
+      return 'processRan';
+    } else if (subType === 'termination_event') {
+      return 'processTerminated';
+    } else {
+      return 'unknownProcessEvent';
+    }
+  } else if (type === 'alert_event') {
+    return 'processCausedAlert';
+  }
+  return 'unknownEvent';
+}
+
+function uniquePidForProcess(event) {
+  return event.data_buffer.node_id;
+}
+
+function uniqueParentPidForProcess(event) {
+  return event.data_buffer.source_id;
+}
+
+function yHalfWayBetweenSourceAndTarget(sourcePosition: Vector2, targetPosition: Vector2) {
```

Comment:
:question: For readability, explaining some of these terms like "source" and "target" would make understanding this for a new reader easier